Assets/Scripts/CombatZone.cs

```diff
@@ -33,50 +38,97 @@ public void Start()
 
     }
 
+    public void Update()
+    {
+        if (!Timer.IsInProgress())
+            return;
+
+        if (CombatStarted)
+            DisablePlayerMovement();
+
+        if (!Timer.IsFinished())
+            SetCombatPositionsUpdate();
+    }
+
     void OnTriggerEnter(Collider other)
     {
         if (CombatStarted)
             return;
 
         if (other.gameObject.CompareTag("Player"))
         {
+            Timer.StartTimer(MovementTime);
+            StartCoroutine(StartingCombat());
```

Comment:
Instead of this, just check in the Update method when `Timer.IsFinished()`. Then, call `Timer.ResetTimer()` to reset it

Reply:
Fixed with ma boi Daniel